### PR TITLE
Windows

### PR DESCRIPTION
**Pull request by pidydx**
[Review on Reviewable](https://reviewable.io/reviews/forensicartifacts/artifacts/62)

### CODE REVIEW

File: definitions/windows.yaml [hunk 1]
```diff
@@ -643,3 +647,208 @@ sources:
...
+- type: FILE
+  attributes:
+    paths:
+      - '%%environ_windir%%\winstart.bat'
+      - '%%environ_windir%%\dosstart.bat'
+    separator: '\'
+supported_os: [Windows]
+---
+name: WindowsIniFiles
+doc: Windows ini file
```
Comment (joachimmetz) [inline definitions/windows.yaml]: This is a bit generic name I opt:
WindowsSystemIniFiles or equiv

File: definitions/windows.yaml [hunk 2]
```diff
@@ -643,3 +647,208 @@ sources:
 - type: WMI
   attributes: {query: SELECT * FROM Win32_Service}
 supported_os: [Windows]
+---
+name: WindowsAutoexec
```
Comment (joachimmetz) [inline definitions/windows.yaml]: WindowsAutoexec => WindowsAutoexecBat

File: definitions/windows.yaml [hunk 3]
```diff
@@ -643,3 +647,208 @@ sources:
...
+    separator: '\'
+supported_os: [Windows]
+---
+name: WindowsSessionManagerSetupExecute
+doc: Windows Session Manager SetupExecute persistence
+sources:
+- type: REGISTRY_VALUE
+  attributes:
+    key_value_pairs: [{key: 'HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager', value: 'SetupExecute'}]
+supported_os: [Windows]
```
Comment (joachimmetz) [inline definitions/windows.yaml]: + URL
https://msdn.microsoft.com/en-us/library/windows/desktop/dd392286%28v=vs.85%29.aspx ?

File: definitions/windows.yaml [hunk 4]
```diff
@@ -643,3 +647,208 @@ sources:
...
+sources:
+- type: REGISTRY_VALUE
+  attributes:
+    key_value_pairs:
+      - {key: 'HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Windows', value: 'load'}
+      - {key: 'HKEY_USERS\%%users.sid%%\Software\Microsoft\Windows NT\CurrentVersion\Windows', value: 'load'}
+      - {key: 'HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Windows', value: 'run'}
+      - {key: 'HKEY_USERS\%%users.sid%%\Software\Microsoft\Windows NT\CurrentVersion\Windows', value: 'run'}
+supported_os: [Windows]
+urls: http://support.microsoft.com/en-us/kb/103865
```
Comment (joachimmetz) [inline definitions/windows.yaml]: urls: ['...']

**Comment by joachimmetz** [comment]
Some nits. Also please add links to e.g. MSDN or related context where you can.

File: definitions/windows.yaml (hunk 2 above)
Comment (pidydx) [inline definitions/windows.yaml]: Done

File: definitions/windows.yaml (hunk 3 above)
Comment (pidydx) [inline definitions/windows.yaml]: Done

File: definitions/windows.yaml (hunk 1 above)
Comment (pidydx) [inline definitions/windows.yaml]: Done

File: definitions/windows.yaml [hunk 5]
```diff
@@ -643,3 +647,217 @@ sources:
...
+sources:
+- type: REGISTRY_VALUE
+  attributes:
+    key_value_pairs:
+      - {key: 'HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager', value: 'PendingFileRenameOperations'}
+supported_os: [Windows]
+urls: ['https://technet.microsoft.com/en-us/library/cc960241.aspx']
+---
+name: WindowsMSDTCDlls
+doc: Windows MSDTC attempts to load these dlls on start
```
Comment (joachimmetz) [inline definitions/windows.yaml]: dlls => DLLs

File: definitions/windows.yaml [hunk 6]
```diff
@@ -643,3 +647,217 @@ sources:
...
+doc: Windows shell replacement
+sources:
+- type: REGISTRY_VALUE
+  attributes:
+    key_value_pairs:    
+      - {key: 'HKEY_USERS\%%users.sid%%\Software\Policies\Microsoft\Windows\System\Scripts', value: 'Logon'}
+      - {key: 'HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows\System\Scripts', value: 'Logon'}
+supported_os: [Windows]
+urls: ['https://technet.microsoft.com/en-us/library/cc940158.aspx']
+
```
Comment (joachimmetz) [inline definitions/windows.yaml]: - white line

File: definitions/windows.yaml [hunk 7]
```diff
@@ -643,3 +647,217 @@ sources:
...
+sources:
+- type: REGISTRY_VALUE
+  attributes:
+    key_value_pairs:
+      - {key: 'HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon', value: 'VmApplet'}
+supported_os: [Windows]
+urls: ['https://technet.microsoft.com/en-us/library/cc939701.aspx']
+---
+name: WindowsStartupScript
+doc: Windows shell replacement
```
Comment (joachimmetz) [inline definitions/windows.yaml]: doc does not match name

Maybe also add the URL: https://technet.microsoft.com/en-us/library/ff404236.aspx

File: definitions/windows.yaml [hunk 8]
```diff
@@ -643,3 +647,217 @@ sources:
...
+sources:
+- type: REGISTRY_VALUE
+  attributes:
+    key_value_pairs:    
+      - {key: 'HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows\System\Scripts', value: 'Startup'}
+supported_os: [Windows]
+urls: ['https://technet.microsoft.com/en-us/library/cc940158.aspx']
+---
+name: WindowsLogonScript
+doc: Windows shell replacement
```
Comment (joachimmetz) [inline definitions/windows.yaml]: change doc

**Comment by joachimmetz** [comment]
Some doc strings that are copies.

File: definitions/windows.yaml (hunk 7 above)
Comment (pidydx) [inline definitions/windows.yaml]: This is a much better URL, thanks!

File: definitions/windows.yaml (hunk 5 above)
Comment (pidydx) [inline definitions/windows.yaml]: Done

File: definitions/windows.yaml (hunk 6 above)
Comment (pidydx) [inline definitions/windows.yaml]: Done

File: definitions/windows.yaml (hunk 8 above)
Comment (pidydx) [inline definitions/windows.yaml]: Done

**Comment by joachimmetz** [comment]
Waiting for the tests but otherwise LGTM

**Pull request merged**